packages/build-tools/src/utils/fingerprint.ts

```diff
@@ -0,0 +1,180 @@
+/**
+ * DO NOT EDIT unless the same change is made in `@expo/fingerprint`
+ * The diffFingerprints function is a copy/paste from https://github.com/expo/expo/pull/29709
+ */
+
+export type FingerprintSource = HashSource & {
+  /**
+   * Hash value of the `source`.
+   * If the source is excluding by `Options.dirExcludes`, the value will be null.
+   */
+  hash: string | null;
+  /**
+   * Debug info from the hashing process. Differs based on source type. Designed to be consumed by humans
+   * as opposed to programmatically.
+   */
+  debugInfo?: DebugInfo;
+};
+
+export interface Fingerprint {
+  /**
+   * Sources and their hash values to generate a fingerprint
+   */
+  sources: FingerprintSource[];
+
+  /**
+   * The final hash value of the whole fingerprint
+   */
+  hash: string;
+}
```

Comment:
Can we import types like this from `@expo/fingerprint`? I'm asking because it seems to me that it might be easier to keep types/functions in sync just by bumping the package version (assuming the package will be backward compatible).

Reply:
Discussed importing this function with @Kudo and the summary of the discussion is that we can't because we don't want to add `@expo/fingerprint` as a dependency.